cogboard-webapp/src/components/ComponentIdDisplay.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import TextField from '@material-ui/core/TextField';
+import Tooltip from '@material-ui/core/Tooltip';
+import FingerprintIcon from '@material-ui/icons/Fingerprint';
+import * as PropTypes from 'prop-types';
+import styled from '@emotion/styled/macro';
+
+const HideableTextField = styled(TextField)`
+  width: 100px;
+  height: 32px;
+
+  &.hidden {
+    display: none;
+    visibility: hidden;
+  }
+`;
+
+const StyledContainer = styled('span')`
+  float: right;
+
+  & .MuiFormControl-marginDense {
+    margin-top: 0;
+    margin-bottom: 0;
+  }
+`;
```

Comment:
Styling is a bit crude, but the main objective was to make it work, and then make it beautiful, turned out pretty nice IMO.